### ISSUE TITLE
False positive: cannot pattern-match on an array without a fixed length E0730

### ISSUE THREAD

**Issue by KizzyCode** [rust-lang/rust-analyzer#23133]
If a specific cargo target is set in the workspace config, rust analyzer produces a
> cannot pattern-match on an array without a fixed length E0730

false positive. However, cargo build works flawlessly and the resulting firmware also works. I have attached a screenshot and the minimal reproducer.

**rust-analyzer version**: `rust-analyzer version: 0.3.3008-standalone (f938641be5 2026-08-10)`

**rustc version**: `rustc 1.97.1 (8bab26f4f 2026-07-14)`

**editor or extension**: VSCode

**relevant settings**: In the workspace settings, `rust-analyzer.cargo.target` must be set to a specific target; e.g.:
```json
// This is `./vscode/settings.json`
{
    "rust-analyzer.cargo.target": "thumbv7em-none-eabihf",
}
```

**code snippet to reproduce**:
```rust
// This is `./src/main.rs`

const MAX_PAYLOAD_DATA_LENGTH: usize = 255;

pub struct PayloadPacket {
    data: [u8; MAX_PAYLOAD_DATA_LENGTH],
}
impl PayloadPacket {
    pub fn payload(&self) -> &[u8] {
        let [_, _, encoded_payload @ ..] = &self.data; // <- this line fails
        encoded_payload
    }
}

fn main() -> ! {
    todo!()
}
```

<img width="1450" height="783" alt="Image" src="https://github.com/user-attachments/assets/19e70245-d43f-48ab-bd09-3cc7d8d02adc" />

[reproducer.zip](https://github.com/user-attachments/files/31007581/reproducer.zip)
